### PR TITLE
Update login UI

### PR DESCRIPTION
**Pull request by CannonLock**
- Memory Cache User and other information
- Remove Jumpiness from various pages that rely on this now cached information
- Add a User button
- Rework the login page to push towards Oauth login

### CODE REVIEW

File: web_ui/frontend/app/(login)/login/page.tsx
Comment (haoming29) [inline web_ui/frontend/app/(login)/login/page.tsx:127]:
```suggestion
                    <span>Login</span>
```

I think Brian wants this to be `Login` or `Next`

File: web_ui/frontend/app/(login)/login/page.tsx
Comment (haoming29) [inline web_ui/frontend/app/(login)/login/page.tsx:174]:
```suggestion
                                    href={"/api/v1.0/auth/oauth/login?next_url=/"}
```

I updated the endpoint in #963 
Since Origin admin will also be able to login via OAuth, we should redirect user to `/` after they are authenticated, or even better, we should redirect them to where they were before hitting the login page

File: web_ui/frontend/app/(login)/login/page.tsx
Comment (haoming29) [inline web_ui/frontend/app/(login)/login/page.tsx:133]: We should include both `origin` and `cache` besides registry for OAuth login

File: web_ui/frontend/app/registry/layout.tsx
Comment (haoming29) [inline web_ui/frontend/app/registry/layout.tsx:1]: Currently, when I'm in "register an origin" or "register a cache" page and hit the Pelican logo, I was redirected to `/`. I was expected to be redirected to `/registry` instead.

File: web_ui/frontend/app/registry/page.tsx
Comment (haoming29) [inline web_ui/frontend/app/registry/page.tsx:106]: Is it possible to have a redirect query parameter here to redirect user back to registration page?

File: web_ui/frontend/components/layout/AuthenticatedContent.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/AuthenticatedContent.tsx:64]: Same here, can we take query param to redirect user back to where they were once they are authenticated?

File: web_ui/frontend/components/layout/Sidebar/RandomUserIcon.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/Sidebar/RandomUserIcon.tsx:8]: I noticed the jumpiness of the user icon every time I refresh/hit a new page and I'm pretty confused. Can we fixed it to a generic user icon?

File: web_ui/frontend/components/layout/Sidebar/UserMenu.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:68]: Can we have a hover-over tooltip describing this icon?

File: web_ui/frontend/components/layout/Sidebar/UserMenu.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:83]: Can we have a hover-over tooltip here as well? We can say "Admin User" | "General User" or just "Profile"

Comment (haoming29) [reply web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:83]: Also, since we display the user status anyway, do we want to distinguish between admin user and general user by the UserIcon (besides having "admin" as the menu item below)?

File: web_ui/frontend/components/layout/Sidebar/UserMenu.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:101]: Can we do a hover-over tooltip explaining what this means? something like "You have administrator privileges" or "You are an admin of this service" or changing the name to "Admin User"

File: web_ui/frontend/components/layout/Sidebar/index.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/Sidebar/index.tsx:3]:
```suggestion
 * Copyright (C) 2024, Pelican Project, Morgridge Institute for Research
```

**Review by haoming29 — CHANGES_REQUESTED** [review]
Some changes to consider (see comments above)

File: web_ui/frontend/app/registry/layout.tsx
Comment (CannonLock) [reply web_ui/frontend/app/registry/layout.tsx:1]: @haoming29 Are you running with the nginx router? Any request to / should redirect you to /registry

Comment (haoming29) [reply web_ui/frontend/app/registry/layout.tsx:1]: No I didn't. I ran it with the built binary

File: web_ui/frontend/components/layout/AuthenticatedContent.tsx
Comment (CannonLock) [reply web_ui/frontend/components/layout/AuthenticatedContent.tsx:64]: This should, did you find a case where it didn't?

Comment (haoming29) [reply web_ui/frontend/components/layout/AuthenticatedContent.tsx:64]: When I hit the button to register an origin at the registry, it redirects me to login page, but after I logged in I was redirected to the registry page

Comment (haoming29) [reply web_ui/frontend/components/layout/AuthenticatedContent.tsx:64]: Same thing with the config page. I used admin password for auth

File: web_ui/frontend/app/registry/layout.tsx
Comment (CannonLock) [reply web_ui/frontend/app/registry/layout.tsx:1]: Are you doing pelican in a box?

**Comment by CannonLock** [comment]
@haoming29 Going to have to move the lack of returning to the original place to next release. I am not sure why my implementation for this is not working post build.

**Comment by haoming29** [comment]
> @haoming29 Going to have to move the lack of returning to the original place to next release. I am not sure why my implementation for this is not working post build.

That sounds good to me

File: web_ui/frontend/app/(login)/login/page.tsx
Comment (haoming29) [inline web_ui/frontend/app/(login)/login/page.tsx:133]: Sorry I was misreading the code. I was meant to suggest that admin password login should be available to all server UIs, but the **OAuth** should be available for registry, origin, and cache. I'll let you decide what to push in this condition here

File: web_ui/frontend/app/(login)/login/page.tsx
Comment (haoming29) [inline web_ui/frontend/app/(login)/login/page.tsx:177]: same here

File: web_ui/frontend/components/layout/Sidebar/UserMenu.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:90]:
```suggestion
                { user.role === "admin" ? <AdminPanelSettings/>: < AccountCircleIcon /> }
```

`Face2` is an icon of a women, and I don't want to make such assumption for a general user, `AccountCircleIcon` seems to be more gender-neutral.

File: web_ui/frontend/components/layout/Sidebar/UserMenu.tsx
Comment (haoming29) [inline web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:7]:
```suggestion
import {Login, AccountCircleIcon, CloudSync, AdminPanelSettings} from "@mui/icons-material";
```

**Review by haoming29 — CHANGES_REQUESTED** [review]
Two small changes requested. We one step before are getting there!

File: web_ui/frontend/components/layout/Sidebar/UserMenu.tsx
Comment (CannonLock) [reply web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:90]: Bah, the logical choice :) I thought it would be fun to make it a woman, but you are right that we should pick something gender neutral.

Comment (haoming29) [reply web_ui/frontend/components/layout/Sidebar/UserMenu.tsx:90]: > Bah, the logical choice :) I thought it would be fun to make it a woman, but you are right that we should pick something gender neutral.

I actually like the random user icon once I realized what's happening, but not sure if bosses will like them 🤔

**Comment by haoming29** [comment]
The cache PR causes a bit merge conflict, but after fixing that, we should hold until the login API changes #963  are in

**Comment by CannonLock** [comment]
@haoming29 Rebased and addressed concerns listed here: https://github.com/PelicanPlatform/pelican/pull/963#issuecomment-2040230213

**Review by haoming29 — APPROVED** [review]
LGTM! Thanks for this

**Comment by haoming29** [comment]
Note that we still need wait until #963 to be merged in first

**Comment by haoming29** [comment]
@CannonLock We should be good to go, once the branch is rebased/merged against the main and the merge conflict are fixed

**Pull request merged**
